File: apps/d-day/d-day-service/src/main/java/com/booster/dday/config/WebConfig.java
package com.booster.dday.config;

import com.booster.dday.shared.web.CurrentMemberArgumentResolver;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.List;

/**
 * {@code @CurrentMember} 를 쓸 수 있게 한다.
 *
 * <p>컨트롤러마다 {@code @RequestHeader("X-User-Id")} 를 적고 널 검사를 하면
 * <b>그 검사를 한 군데서 빠뜨리는 날 개인 자원이 열린다.</b> 해석기 하나로 모아 두면
 * 빠뜨릴 자리가 없다.
 */
@Configuration
@RequiredArgsConstructor
public class WebConfig implements WebMvcConfigurer {

    private final CurrentMemberArgumentResolver currentMemberArgumentResolver;

    @Override
    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
        resolvers.add(currentMemberArgumentResolver);
    }
}

File: apps/kotlin/damagochi-service/src/main/kotlin/com/booster/kotlin/damagochiservice/common/web/WebConfig.kt
package com.booster.kotlin.damagochiservice.common.web

import org.springframework.context.annotation.Configuration
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig(
    private val currentUserIdResolver: CurrentUserIdResolver
) : WebMvcConfigurer {
    override fun addArgumentResolvers(resolvers: MutableList<HandlerMethodArgumentResolver>) {
        resolvers.add(currentUserIdResolver)
    }

    override fun addCorsMappings(registry: CorsRegistry) {
        registry.addMapping("/api/**")
            .allowedOriginPatterns(
                "http://localhost:*",
                "http://127.0.0.1:*"
            )
            .allowedMethods("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS")
            .allowedHeaders("*")
            .exposedHeaders("*")
            .allowCredentials(false)
            .maxAge(3600)
    }
}



